### PR TITLE
Increased net flares to 10

### PR DESCRIPTION
**Pull request by tritonas00**
and bumped protocol version to 2.43

goes by hand with https://github.com/RigsOfRods/ror-server/pull/109

closes https://github.com/RigsOfRods/rigs-of-rods/issues/2664

not tested yet

### CODE REVIEW

**Comment by only-a-ptr** [comment]
TODO:
- [ ] in "SimConstants.h", add MAX_CLIGHTS=10 with comment // limited for networking
- [ ] enlarge `Actor::m_net_custom_lights` to MAX_CLIGHTS and make it `int`;
- [ ] in `ActorSpawner::InitializeRig`, put `for (int i=0; i<MAX_CLIGHTS; ++i) { m_net_custom_lights = -1; }`, remove existing code.
- [ ] in `Actor::[get/set]CustomLightVisible()`, fix up the range check and simplify the code to `m_net_custom_lights[number] != -1` and reading/setting `ar_flares[m_net_custom_lights[number]].controltoggle_status`

**Comment by tritonas00** [comment]
Done (multiple facepalms incoming... :laughing:)

File: source/main/physics/Actor.cpp [hunk 1]
```diff
@@ -4523,30 +4523,26 @@ bool Actor::getBrakeLightVisible()
 
 bool Actor::getCustomLightVisible(int number)
 {
-    if (number < 0 || number > 3)
+    if (number < 0 || number > 9)
```
Comment (only-a-ptr) [inline source/main/physics/Actor.cpp:4526]: nitpick: `if (number < 0 || number >= MAX_CLIGHTS)`

File: source/main/physics/Actor.cpp [hunk 2]
```diff
@@ -4523,30 +4523,26 @@ bool Actor::getBrakeLightVisible()
...
     {
-        LOG("AngelScript: Invalid Light ID (" + TOSTRING(number) + "), allowed range is (0 - 4)");
+        LOG("AngelScript: Invalid Light ID (" + TOSTRING(number) + "), allowed range is (0 - 9)");
```
Comment (only-a-ptr) [inline source/main/physics/Actor.cpp:4528]: remake pls: `fmt::format("invalid custom-light ID {}, allowed range is 0-{}", number, MAX_CLIGHTS-1)`

**Review by only-a-ptr — APPROVED** [review]
Nice work

**Comment by tritonas00** [comment]
@only-a-ptr 

https://github.com/RigsOfRods/rigs-of-rods/pull/2690/files#diff-2b5d30a8bd6eb8a53fefb2a3c3bf32ec559db32ba884c654805afdede400dc5bR4544

maybe it should be `m_net_custom_lights[number] < ar_flares.size()` ?

**Comment by only-a-ptr** [comment]
New requirement based on https://github.com/RigsOfRods/rigs-of-rods/issues/2664#issuecomment-782053976

* delete `int m_net_custom_lights_count`, make it `bool m_net_custom_lights_state[MAX_CLIGHTS]`
* in spawner, assign 'false' instead of -1 in the init loop. Delete all other code.
* in `Actor::UpdateFlareStates(float dt)`, under the `ar_flares[i].controltoggle_status = ! ar_flares[i].controltoggle_status;` line, add 'm_net_custom_light_state[ar_flares[i].controlnumber - 1] = ar_flares[i].controltoggle_status;'
* change the `[get/set]CustomLightVisible` to just read/write `m_net_custom_light_state[number]`

That should do the trick.

**Comment by tritonas00** [comment]
Done (i think :smile:)

**Comment by only-a-ptr** [comment]
Replaced by #2729

**Pull request closed without merging**